sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,184 @@
+                ensure!(
+                    config.frequency > 0,
+                    error::InvalidNetConfigSnafu {
+                        reason: "miimon-frequency of 0 disables Mii Monitoring, either set a value or configure Arp Monitoring"
+                    }
+                );
+                // updelay and downdelay should be a multiple of frequency, but will be rounded down
+                // by the kernel, this ensures they are at least the size of frequency (non-zero)
+                ensure!(
+                    config.frequency <= config.updelay && config.frequency <= config.downdelay,
+                    error::InvalidNetConfigSnafu {
+                        reason: "miimon-updelay and miimon-downdelay must be equal to or larger than miimon-frequency"
+                    }
+                )
+            }
+            BondMonitoringConfig::ArpMon(config) => {
+                ensure!(
+                    config.interval > 0,
+                    error::InvalidNetConfigSnafu {
+                        reason: "arpmon-interval of 0 disables Arp Monitoring, either set a value or configure Mii Monitoring"
+                    }
+                );
+                // If using Arp Monitoring, 1-16 targets must be specified
+                let targets_length: u32 = config.targets.len() as u32;
+                ensure!(
+                    targets_length > 0 && targets_length <= 16,
+                    error::InvalidNetConfigSnafu {
+                        reason: "arpmon-targets must include between 1 and 16 targets"
+                    }
+                )
+            }
+        }
+
+        Ok(())
+    }
+}
+
+// Currently mode 1 (active-backup) is only supported but eventually 0-6 could be added
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) enum BondMode {
+    #[serde(rename = "active-backup")]
+    ActiveBackup,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum BondMonitoringConfig {
+    MiiMon(MiiMonitoringConfig),
+    ArpMon(ArpMonitoringConfig),
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct MiiMonitoringConfig {
+    #[serde(rename = "miimon-frequency")]
+    pub(crate) frequency: u32,
+    #[serde(rename = "miimon-updelay")]
+    pub(crate) updelay: u32,
+    #[serde(rename = "miimon-downdelay")]
+    pub(crate) downdelay: u32,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct ArpMonitoringConfig {
+    #[serde(rename = "arpmon-interval")]
+    pub(crate) interval: u32,
+    #[serde(rename = "arpmon-validate")]
+    pub(crate) validate: ArpValidate,
+    #[serde(rename = "arpmon-targets")]
+    pub(crate) targets: Vec<IpAddr>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) enum ArpValidate {
+    #[serde(rename = "active")]
+    Active,
+    #[serde(rename = "all")]
+    All,
+    #[serde(rename = "backup")]
+    Backup,
+    #[serde(rename = "none")]
+    None,
+}
```

Comment:
(or lowercase...)

This is generally applicable anywhere you have used `[serde(rename="whatever")]` when it could actually be done with `serde(rename_all = "kebab-case")`

```suggestion
#[derive(Clone, Debug, Deserialize)]
#[serde(rename_all = "kebab-case")]
pub(crate) enum ArpValidate {
    Active,
    All,
    Backup,
    None,
}
```

Reply:
Yeah, this is much nicer, I'll change to this.